src/state/Maps/selectors.js

```diff
@@ -1,1181 +1,482 @@
-	(layers, layerKey) => {
-		if (layers && layerKey) {
-			return layers.find(l => l.key === layerKey)
-		} else {
-			return null;
-		}
-	}
+const getMapSetLayersStateByMapKey = createSelector(
+    [
+        getMapSetByMapKey
+    ],
+    (set) => {
+        return set?.data?.layers || null;
+    }
 );
 
-
-
-
-
-/* ===== Combined selectors ==================== */
-
 /**
  * @param state {Object}
  * @param mapKey {string}
  */
-const getFilterByActiveByMapKey = createCachedSelector(
-	[
-		getMapFilterByActiveByMapKey,
-		getMapSetFilterByActiveByMapKey
-	],
-	(mapFilter, setFilter) => {
-		return (setFilter || mapFilter) && {...setFilter, ...mapFilter};
-	}
-)((state, mapKey) => mapKey);
+const getMapMetadataModifiersByMapKey = createSelector(
+    [
+        getMapByKey
+    ],
+    (map) => {
+        return map?.data?.metadataModifiers || null;
+    }
+);
 
 /**
  * @param state {Object}
  * @param mapKey {string}
  */
-const getBackgroundLayerStateByMapKey = createCachedSelector(
-	[
-		getMapBackgroundLayerStateByMapKey,
-		getMapSetBackgroundLayerStateByMapKey,
-	],
-	(mapBackgroundLayer, setBackgroundLayer) => {
-		return mapBackgroundLayer || setBackgroundLayer;
-	}
-)(
-	(state, mapKey) => `${mapKey}`
+const getMapSetMetadataModifiersByMapKey = createSelector(
+    [
+        getMapSetByMapKey
+    ],
+    (set) => {
+        return set?.data?.metadataModifiers || null;
+    }
 );
 
-
 /**
  * @param state {Object}
  * @param mapKey {string}
- * @return {Object} Merged layer state from mapState and mapSetState with metadataModifiers and filterByActive.
  */
-const getLayersStateByMapKey = createCachedSelector(
-	[
-		getMapLayersStateByMapKey,
-		getMapSetLayersStateByMapKey,
-		getMapMetadataModifiersByMapKey,
-		getMapSetMetadataModifiersByMapKey,
-		getFilterByActiveByMapKey
-	],
-	(mapLayers, setLayers, mapMetadataModifiers, setMetadataModifiers, mapFilterByActive) => {
-		if (mapLayers || setLayers) {
-			let layers = [...(setLayers || []), ...(mapLayers || [])];
-			let modifiers = {};
-			if (setMetadataModifiers) {
-				modifiers = setMetadataModifiers;
-			}
-			modifiers = {...modifiers, ...mapMetadataModifiers};
-
-			layers = layers.map(layer => {
-				let layerMetadataModifiers = layer.metadataModifiers ? {...modifiers, ...layer.metadataModifiers} : modifiers;
-				let layerFilterByActive = layer.filterByActive ? {...mapFilterByActive, ...layer.filterByActive} : mapFilterByActive;
-
-				return {...layer, metadataModifiers: layerMetadataModifiers, filterByActive: layerFilterByActive};
-			});
-
-			// TODO return error for duplicates?
-			return _.uniqBy(layers, 'key');
-		} else {
-			return null;
-		}
-	}
+const getMetadataModifiersByMapKey = createCachedSelector(
+    [
+        getMapMetadataModifiersByMapKey,
+        getMapSetMetadataModifiersByMapKey
+    ],
+    (mapModifiers, setModifiers) => {
+        if (mapModifiers && setModifiers) {
+            return {...mapModifiers, ...setModifiers};
```

Comment:
Are set modifiers to override map modifiers in case of conflict?

Reply:
You're right, it should be switched. Fixed: https://github.com/gisat-panther/ptr-state/pull/48/commits/c65e82673e93df6d9bc57aa5486ec0a2bb91faab